### PR TITLE
task 5 global exception handler build success

### PR DESCRIPTION
**Pull request by MolchanovAlexander**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
lgfm!

File: src/main/java/com/example/bookstorewebapp/controller/BookController.java
Comment (makaroshyna) [inline src/main/java/com/example/bookstorewebapp/controller/BookController.java:1]: You could also add @Valid to updateBook method.

**Review by makaroshyna — APPROVED** [review]
GJ

File: src/main/java/com/example/bookstorewebapp/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
 @Data
 public class CreateBookRequestDto {
+    @NotBlank
+    @Size(min = 2, max = 255)
     private String title;
+    @NotBlank
+    @Size(min = 2, max = 255)
     private String author;
+    @NotBlank
+    @Pattern(regexp = "^\\d{13}$", message = "The ISBN must contain 13 digits")
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstorewebapp/dto/CreateBookRequestDto.java]: I believe there is specific annotation to validate ISBN

File: src/main/java/com/example/bookstorewebapp/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    private static final String ERROR = "errors";
+    private static final String DELIMITER = " ";
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request
+    ) {
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstorewebapp/exception/CustomGlobalExceptionHandler.java:32]: Please add methods to handle other exceptions that could occur in your code

File: src/main/java/com/example/bookstorewebapp/repository/book/BookSpecificationBuilder.java [hunk 3]
```diff
@@ -14,8 +13,7 @@
 public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
     private static final String TITLE = "title";
     private static final String AUTHOR = "author";
-    @Autowired
-    private SpecificationProviderManager<Book> bookSpecificationProviderManager;
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstorewebapp/repository/book/BookSpecificationBuilder.java:16]: Is this changed related to task?

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good job, but some improvements required

File: src/main/java/com/example/bookstorewebapp/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
 @Data
 public class CreateBookRequestDto {
+    @NotBlank
+    @Size(min = 2, max = 255)
     private String title;
+    @NotBlank
+    @Size(min = 2, max = 255)
     private String author;
+    @NotBlank
+    @Isbn
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookstorewebapp/dto/CreateBookRequestDto.java:20]: there is `@ISBN` in hibernate validation https://docs.jboss.org/hibernate/stable/validator/api/org/hibernate/validator/constraints/ISBN.html

but you can leave custom if you like

File: src/main/java/com/example/bookstorewebapp/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,13 +1,26 @@
...
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotBlank
+    @Size(min = 2, max = 255)
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookstorewebapp/dto/CreateBookRequestDto.java:14]: adding messages to the constraints would be great

**Review by yevheniisynytsia — APPROVED** [review]

File: src/main/java/com/example/bookstorewebapp/dto/CreateBookRequestDto.java (hunk 5 above)
Comment (MolchanovAlexander) [reply src/main/java/com/example/bookstorewebapp/dto/CreateBookRequestDto.java:14]: But they have it inside by default and add to response values like name must have from 2 to 255 symbols.

**Pull request merged**